convert2rhel/actions/handle_packages.py

```diff
@@ -0,0 +1,110 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+
+from convert2rhel import actions, pkghandler, pkgmanager
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+
+class ListThirdPartyPackages(actions.Action):
+    dependencies = tuple()
+    severity = actions.SEVERITY_LEVELS["WARNING"]
+
+    def run(self):
+        logger.task("Convert: List third-party packages")
+        pkghandler.list_third_party_pkgs()
+
+
+class _PackageRemoval:
+    def remove_packages_with_confirmation(self, packages):
+        pkghandler.remove_pkgs_with_confirm(packages)
+
+
+class RemoveExcludedPackages(actions.Action, _PackageRemoval):
```

Comment:
I think I misunderstood a question you were asking the other day.  I would probably call `pkghandler.remove_pkgs_with_confirm(packages)` directly from the Action class.  Since it's just a single function call, I don't think we gain much by putting it into a function and a class inside of handle_packages.  Does that make sense?

If you do think it should be grouped with `handle_packages`, I would move it out to module level.  When helper functons don't make use of `self`, we should usually leave them at module scope.  In other languages, classes are the natural way to create namespaces but in Python modules are the natural way to create namespaces so classes should only hold values if they are really needed there.
```suggestion
def _remove_packages_with_confirmation(packages):
    pkghandler.remove_pkgs_with_confirm(packages)


class RemoveExcludedPackages(actions.Action):
```

Note: If you do decide it should remain inside of a class, then you should know that the method resolution order is left to right.  That means if you have one class that's obviously the "base" for that class hierarchy and other classes which are mixins which are there to add specific traits to this particular class, then you would want the mixins to be to the left and the base to be the furthest right.  Mixins should probably also always call `super()` so they don't entirely mask any methods of the same name which could be added to other parents later. So, if you didn't change this to be a toplevel function, it would look like this:
``` python
class _PackageRemoval:
    def remove_packages_with_confirmation(self, packages):
        pkghandler.remove_pkgs_with_confirm(packages)
        super(_PackageRemoval, self).remove_packages_with_confirmaton(packages)


class Remove ExcludedPackages(_PackageRemoval, actions.Action):
```

Reply:
I revisited that today and I think that the extra class/function will not be necessary at all... Having the `remove_pkgs_with_confirm()` directly in the action class run method is way better than adding this another layer on top of it.